### PR TITLE
backupccl: clean up old restore syntax code

### PR DESCRIPTION
**Pull request by kev-cao**
The old restore syntax was fully removed in #133610. This commit removes all the old code that ran for the old syntax.

Epic: none

Release note: none

### CODE REVIEW

File: pkg/sql/sem/tree/backup.go [hunk 1]
```diff
@@ -157,20 +157,16 @@ type Restore struct {
...
-	//   - len(From)>1 implies the user explicitly passed incremental backup paths,
-	//     which is only allowed using the old syntax, `RESTORE <targets> FROM <destination>.
-	//     In this case, From[0] contains the URI(s) for the full backup.
-	//   - len(From)==1 implies we'll have to look for incremental backups in planning
-	//   - len(From[0]) > 1 implies the backups are locality aware
-	//   - From[i][0] must be the default locality.
-	From    []StringOrPlaceholderOptList
+	// From contains the URIs for the backup we seek to restore.
+	//   - len(From) > 1 implies the backups are locality aware
+	//   - From[0] must be the default locality.
```
Comment (msbutler) [inline pkg/sql/sem/tree/backup.go:162]: music to my ears.

File: pkg/ccl/backupccl/backupdest/backup_destination.go [hunk 2]
```diff
@@ -638,82 +638,3 @@ func ResolveBackupManifests(
 	}
 	return validatedDefaultURIs, validatedMainBackupManifests, validatedLocalityInfo, totalMemSize, nil
 }
-
```
Comment (msbutler) [inline pkg/ccl/backupccl/backupdest/backup_destination.go:641]: excellent

**Review by msbutler — APPROVED** [review]
This pr made my day.

File: pkg/sql/sem/tree/pretty.go [hunk 3]
```diff
@@ -2187,15 +2187,10 @@ func (node *Restore) doc(p *PrettyCfg) pretty.Doc {
 	if node.DescriptorCoverage == RequestedDescriptors {
 		items = append(items, node.Targets.docRow(p))
 	}
```
[the remark this answers is not in the document]
Comment (kev-cao) [reply pkg/sql/sem/tree/pretty.go:2189]: Nope this is not

[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/sql/sem/tree/pretty.go:2189]: why don't we add FROM and IN when subdir is nil?

[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/sql/sem/tree/pretty.go:2189]: oh, i think subdir is always filled, so we can get rid of the subdir !=nil gate here.

[the remark this answers is not in the document]
Comment (kev-cao) [reply pkg/sql/sem/tree/pretty.go:2189]: Oh yeah it looks like it always is filled, updated

**Comment by kev-cao** [comment]
TFTR!

bors r=msbutler

**Pull request merged**